### DISCUSSION TITLE
How to add row number to Datagrid.

### DISCUSSION THREAD

**Discussion by ShrutiJaiswal1494** [AvaloniaUI/Avalonia#6249] [Q&A] [answered]
Hi,

I am looking for a way to display the row no's in the Datagrid as a `row header`. I have added this code to set the row numbers but it's not getting displayed. Is there a way to display the row numbers?
Thanks.

Code
```
m_dataGrid = this.FindControl<DataGrid>(@"dataGrid");
m_dataGrid.LoadingRow += AddRowNumbers;
public void AddRowNumbers(object sender, DataGridRowEventArgs e)
        {
            e.Row.Header = (e.Row.GetIndex() + 1).ToString();
        }
```

**Comment by maxkatz6** [comment, marked as answer]
What Avalonia version and what styles do you use?
In 0.10.0+ it should work. Before - additional styles were required. 

Also, have you set HeadersVisibility property on data grid?

**Reply by ShrutiJaiswal1494** [reply]
I was using 0.10.0+ and fluent style. However, I missed setting the HeadersVisibility property.
